mayastor/src/bdev/nexus/nexus_child_error_store.rs

```diff
@@ -0,0 +1,302 @@
+use spdk_sys::{spdk_bdev, spdk_bdev_io_type};
+use std::{
+    fmt::{Debug, Display},
+    time::{Duration, Instant},
+};
+
+use crate::subsys::Config;
+
+use crate::bdev::nexus::nexus_io::{io_status, io_type};
+
+use crate::core::{Cores, Reactors};
+
+use crate::bdev::nexus::{
+    nexus_bdev,
+    nexus_bdev::{
+        nexus_lookup,
+        Error::{ChildMissing, ChildMissingErrStore},
+        Nexus,
+    },
+};
+
+use serde::export::{fmt::Error, Formatter};
+
+#[derive(Copy, Clone)]
+pub struct NexusChildErrorRecord {
+    io_offset: u64,
+    io_num_blocks: u64,
+    timestamp: Instant,
+    io_error: i32,
+    io_op: spdk_bdev_io_type,
+}
+
+impl Default for NexusChildErrorRecord {
+    fn default() -> Self {
+        Self {
+            io_op: 0,
+            io_error: 0,
+            io_offset: 0,
+            io_num_blocks: 0,
+            timestamp: Instant::now(), // for want of another suitable default
+        }
+    }
+}
+
+pub struct NexusErrStore {
+    no_of_records: usize,
+    next_record_index: usize,
+    records: Vec<NexusChildErrorRecord>,
+}
+
+impl NexusErrStore {
+    pub const READ_FLAG: u32 = 1;
```

Comment:
Rather than declaring these, would it be possible to just reuse the io_type::*? eg 1<<io_type::READ

Reply:
I did think about that. I like the legibility as it is now, but since you brought it up I might change it.